lib/src/modules/task.cc

```diff
@@ -128,11 +136,171 @@ static TaskCommand getTaskCommand(const fs::path &task_executable)
     return TaskCommand { task_executable.string(), { } };
 }
 
+// Converts boost filesystem errors to a task_error object.
+// TODO: Modify this to convert it to a PXP-error instead.
```

Comment:
`response.setBadResultsAndEnd(execution_error);` is generally how we setup to return a PXP error. See https://github.com/puppetlabs/pxp-agent/blob/master/lib/src/request_processor.cc#L126-L152 for where the PXP error is sent.